packages/cyberstorm/src/components/CustomSelect.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { Dispatch, ReactNode, SetStateAction } from "react";
+import styles from "./componentStyles/CustomSelect.module.css";
+import { Button } from "./Button";
+import { CustomSelectItem } from "./CustomSelectItem";
+import { faChevronDown } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import * as Select from "@radix-ui/react-select";
+import { SelectItem } from "@radix-ui/react-select";
+
+export type SelectOption = {
+  value: string;
+  label?: string;
+  leftIcon?: ReactNode;
+};
+
+type SelectProps = {
+  colorScheme?: "default" | "defaultDark" | "primary";
+  defaultOpen?: boolean;
+  icon?: ReactNode;
+  onChange: Dispatch<SetStateAction<string>>;
+  options: SelectOption[];
+  placeholder?: string;
+  value: string;
+};
+
+export const CustomSelect: React.FC<SelectProps> = (props) => {
+  const {
+    colorScheme,
+    defaultOpen,
+    icon,
+    options,
+    onChange,
+    placeholder,
+    value,
+  } = props;
+
+  const selectItemElements = options
+    ? mapSelectData(options, colorScheme)
+    : null;
+
+  return (
+    <div className={styles.root}>
+      <Select.Root
+        defaultOpen={defaultOpen}
+        value={value}
+        onValueChange={onChange}
+        disabled={options.length == 0}
+      >
+        <Select.Trigger asChild>
+          <div>
+            <Button
+              colorScheme={colorScheme}
+              rightIcon={icon}
+              label={
+                options?.find((o) => o.value === value)?.label ?? placeholder
+              }
+            />
+          </div>
+        </Select.Trigger>
+
+        <Select.Portal>
+          <Select.Content
+            className={`${styles.content} ${getContentStyle(colorScheme)}`}
+          >
+            <Select.Group className={styles.group}>
```

Comment:
Same thing here, is nesting everything inside a group by default even if the items don't belong in a group by nature intentional? This will also prevent any additional grouping of items.